sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosTableSchemaInferer.scala

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.CosmosAsyncClient
+import com.azure.cosmos.models.CosmosQueryRequestOptions
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.{ArrayNode, BinaryNode, BooleanNode, DecimalNode, DoubleNode,
+    FloatNode, IntNode, LongNode, NullNode, ObjectNode, TextNode}
+
+
+// scalastyle:off underscore.import
+import org.apache.spark.sql.types._
+import scala.collection.JavaConverters._
+// scalastyle:on underscore.import
+
+// Infers a schema by reading sample data from a source container.
+private object CosmosTableSchemaInferer
+    extends CosmosLoggingTrait {
+    val RAW_JSON_BODY_ATTRIBUTE_NAME = "_rawBody"
+    private val TIMESTAMP_ATTRIBUTE_NAME = "_ts"
+    private val ID_ATTRIBUTE_NAME = "id"
+    private val defaultSchemaForInferenceDisabled = StructType(Seq(
+        StructField(RAW_JSON_BODY_ATTRIBUTE_NAME, StringType),
+        StructField(ID_ATTRIBUTE_NAME, StringType),
+        StructField(TIMESTAMP_ATTRIBUTE_NAME, LongType)
+    ))
+
+    def inferSchema(inferredItems : Seq[ObjectNode]): StructType = {
+        if (inferredItems.isEmpty){
+            // No documents to infer from
+            StructType(Seq())
```

Comment:
Should we return defaultSchemaForInferenceDisabled  instead?

Reply:
I thought so but if you think about it, the collection is empty, so there will be no rows that will be read, the schema won't apply to anything, so the schema won't be used.